lib/vanagon/component/source/rewrite.rb

```diff
@@ -0,0 +1,85 @@
+require 'vanagon/component/source'
+
+class Vanagon
+  class Component
+    class Source
+      # This class has been extracted from Vanagon::Component::Source for the
+      # sake of isolation and in service of its pending removal. Rewrite rules
+      # should be considered deprecated. The removal will be carried out before
+      # Vanagon 1.0.0 is released.
+      class Rewrite
+        @rewrite_rules = {}
+
+        class << self
+          attr_reader :rewrite_rules
+
+          # @deprecated Please use the component DSL method #mirror(<URI>)
```

Comment:
correct, though a deprecation notice will be emitted by some of these methods now.